onnxruntime/core/optimizer/initializer.h

```diff
@@ -57,8 +57,11 @@ class Initializer final {
     return data_.Data<T>();
   }
```

Comment:
Not sure why we don't expose Tensor::DataAsSpan and MutableDataAsSpan here as that would be better than returning these raw pointers.

Optional to add in this PR, but it might simplify some of the usage in NNAPI/CoreML to have at least DataAsSpan as an option.

Reply:
done. But there are too many usages of data<T>() in core/optimizer

Reply:
You don't need to update all those. Just use DataAsSpan for code you're already updating.

Reply:
Okay, I see.